File: OnPremTemplates/DesktopWeb/RemoteWebDriverTestDesktopWeb.java
import java.io.*;
import java.net.*;
import java.util.*;
import java.util.concurrent.TimeUnit;
import org.openqa.selenium.remote.*;

public class RemoteWebDriverWebDemo {

    public static void main(String[] args) throws MalformedURLException, IOException {
        System.out.println("Run started");

        DesiredCapabilities capabilities = new DesiredCapabilities();
        String host = "myHost.perfectomobile.com";
        String token = "myToken";

        capabilities.setCapability("securityToken", token);

        // TODO: Set the Web Machine configuration, for example:
        capabilities.setCapability("platformName", "Windows");
        capabilities.setCapability("platformVersion", "8.1");
        capabilities.setCapability("browserName", "Firefox");
        capabilities.setCapability("browserVersion", "latest");

        // TODO: Name your script
        capabilities.setCapability("scriptName", "RemoteWebDriverTest");

        RemoteWebDriver driver = new RemoteWebDriver(new URL("https://" + host + "/nexperience/perfectomobile/wd/hub/fast"), capabilities);
        driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
        driver.manage().timeouts().pageLoadTimeout(30, TimeUnit.SECONDS);

        try {
            // TODO: write your code here
            driver.get("http://www.perfectomobile.com");

        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try {
                driver.close();

            } catch (Exception e) {
                e.printStackTrace();
            }

            driver.quit();
        }
        System.out.println("Run ended");
    }
}

File: OnPremTemplates/DesktopWeb/RemoteWebDriverTestDesktopWeb.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Diagnostics;
using System.Threading;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Remote;


namespace PerfectoLabSeleniumTestGoogleHomePage
{
    /// <summary>
    /// Summary description for MobileRemoteTest
    /// </summary>
    [TestClass]
    public class RemoteWebDriverTest
    {
        private RemoteWebDriverExtended driver;

        [TestInitialize]
        public void PerfectoOpenConnection()
        {
            // TODO: Set your cloud host and credentials
            DesiredCapabilities capabilities = new DesiredCapabilities();
            var host = "myHost.perfectomobile.com";
            var token = "myToken";

            //TODO: Set the Web Machine configuration, for example:
            capabilities.SetCapability("platformName", "Windows");
            capabilities.SetCapability("platformVersion", "8.1");
            capabilities.SetCapability("browserName", "Chrome");
            capabilities.SetCapability("browserVersion", "Latest");

            capabilities.SetPerfectoLabExecutionId(host);
            capabilities.SetCapability("securityToken", token);

            // TODO: Name your script
            //capabilities.SetCapability("scriptName", "RemoteWebDriverTest");

            var url = new Uri(string.Format("http://{0}/nexperience/perfectomobile/wd/hub/fast", host));
            driver = new RemoteWebDriverExtended(new HttpAuthenticatedCommandExecutor(url), capabilities);
            driver.Manage().Timeouts().ImplicitlyWait(TimeSpan.FromSeconds(30));
            driver.Manage().Timeouts().SetPageLoadTimeout(TimeSpan.FromSeconds(30));
        }

        [TestCleanup]
        public void PerfectoCloseConnection()
        {

            driver.Close();

            // In case you want to download the report or the report attachments, do it here.
            //try
            //{
            //var parameters = new Dictionary<string, object>();
            //driver.ExecuteScript("mobile:execution:close", parameters);
            //driver.DownloadReport(DownloadReportTypes.pdf, "C:\\test\\report");
            //driver.DownloadAttachment(DownloadAttachmentTypes.video, "C:\\test\\report\\video", "flv");
            //driver.DownloadAttachment(DownloadAttachmentTypes.image, "C:\\test\\report\\images", "jpg");
            //}
            //catch (Exception ex)
            //{
            //    Trace.WriteLine(string.Format("Error getting test logs: {0}", ex.Message));
            //}

            driver.Quit();
        }

        [TestMethod]
        public void SearchGoogle()
        {
            // TODO: Write your test here
            driver.Navigate().GoToUrl("http://www.google.com");
        }
    }
}
